src/predict_request_validation_utils.cpp

```diff
@@ -222,8 +265,45 @@ Status RequestValidator<KFSRequest, KFSTensorInputProto, KFSInputTensorIteratorT
         ++bufferId;
     }
     if (it != request.inputs().end()) {
+        currentlyValidatedName = &name;
         return StatusCode::OK;
     }
+    currentlyValidatedName = nullptr;
+    std::stringstream ss;
+    ss << "Required input: " << name;
+    const std::string details = ss.str();
+    SPDLOG_DEBUG("[servable name: {} version: {}] Missing input with specific name - {}", servableName, servableVersion, details);
+    return Status(StatusCode::INVALID_MISSING_INPUT, details);
+}
+template <>
+Status RequestValidator<ovms::InferenceRequest, InferenceTensor, const InferenceTensor*, shape_t>::validateAndGetInput(const InferenceRequest& request, const std::string& name, const InferenceTensor*& it, size_t& bufferId) {
+    if (request.getInput(name.c_str(), &it) != StatusCode::NONEXISTENT_TENSOR) {
+        currentlyValidatedName = &name;
+        const Buffer* buffer = it->getBuffer();
+        const BufferType bufType = buffer->getBufferType();
+        if (bufType < BufferType::OVMS_BUFFERTYPE_CPU || bufType > BufferType::OVMS_BUFFERTYPE_HDDL) {
+            // Remove this when other buffer types are supported
+            if (bufType != BufferType::OVMS_BUFFERTYPE_CPU) {
```

Comment:
This will never execute when buffer type is hddl. It will not enter if branch in 284. It probably should not be nested if but one after another

Reply:
Also add test that specifying hddl (and other unsupported types) fails validation. It must be mising since this code passed